From ff6c459a10b33aa09b6429d553d8cb21e76f20f5 Mon Sep 17 00:00:00 2001
From: Dirk Brandewie <dirk.j.brandewie@intel.com>
Date: Thu, 2 Oct 2014 08:36:45 -0700
Subject: [PATCH 3/6] intel_pstate: Add tracepoint to byt_set_pstate

Used to verify that VID is being set correctly for a requested P state

Signed-off-by: Dirk Brandewie <dirk.j.brandewie@intel.com>
---
 drivers/cpufreq/intel_pstate.c |    2 ++
 include/trace/events/power.h   |   26 ++++++++++++++++++++++++++
 2 files changed, 28 insertions(+)

--- linux.orig/drivers/cpufreq/intel_pstate.c
+++ linux/drivers/cpufreq/intel_pstate.c
@@ -415,6 +415,8 @@ static void byt_set_pstate(struct cpudat
 	if (pstate > cpudata->pstate.max_pstate)
 		vid = cpudata->vid.turbo;
 
+	trace_pstate_byt_set(pstate, vid);
+
 	val |= vid;
 
 	wrmsrl_on_cpu(cpudata->cpu, MSR_IA32_PERF_CTL, val);
--- linux.orig/include/trace/events/power.h
+++ linux/include/trace/events/power.h
@@ -83,6 +83,32 @@ TRACE_EVENT(pstate_sample,
 
 );
 
+TRACE_EVENT(pstate_byt_set,
+
+	TP_PROTO(u32 pstate,
+		u32 vid
+		),
+
+	TP_ARGS(pstate,
+		vid
+		),
+
+	TP_STRUCT__entry(
+		__field(u32, pstate)
+		__field(u32, vid)
+		),
+
+	TP_fast_assign(
+		__entry->pstate = pstate;
+		__entry->vid = vid;
+		),
+
+	TP_printk("pstate=%lu vid=%lu  ",
+		(unsigned long)__entry->pstate,
+		(unsigned long)__entry->vid
+		)
+);
+
 /* This file can get included multiple times, TRACE_HEADER_MULTI_READ at top */
 #ifndef _PWR_EVENT_AVOID_DOUBLE_DEFINING
 #define _PWR_EVENT_AVOID_DOUBLE_DEFINING
